htdocs/compta/prelevement/class/bonprelevement.class.php

```diff
@@ -1180,7 +1231,11 @@ public function create($banque = 0, $agence = 0, $mode = 'real', $format = 'ALL'
 						 * $fac[11] : IBAN
 						 * $fac[12] : frstrcur
 						 */
-						$ri = $this->AddFacture($fac[0], $fac[2], $fac[8], $fac[7], $fac[3], $fac[4], $fac[5], $fac[6], $type);
+						if (!empty($sourcetype)) {
+							$ri = $this->AddFacture($fac[0], $fac[2], $fac[8], $fac[7], $fac[3], $fac[4], $fac[5], $fac[6], $type, $sourcetype);
```

Comment:
You can merge the 2 if into 1 without if, using always $sourcetype as parameter.